### PR TITLE
refactor: extract `get_lr` warning

### PR DESCRIPTION
**Pull request by david20571015**
Extract the `_get_lr_called_within_step` checking in the `get_lr()` of every LRSchedulers.

### CODE REVIEW

**Review by janeyx99 — APPROVED** [review]
thanks for the BE!

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -195,6 +195,15 @@ def step(self, epoch=None):
         self._last_lr = [group["lr"] for group in self.optimizer.param_groups]
 
 
+def _warn_get_lr_called_within_step(lr_scheduler: LRScheduler):
+    if not lr_scheduler._get_lr_called_within_step:
+        warnings.warn(
+            "To get the last learning rate computed by the scheduler, "
+            "please use `get_last_lr()`.",
+            UserWarning,
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:203]:
```suggestion
            UserWarning,
            stacklevel=2
```
We should report the warning on the get_lr call that the user is making, not this internal one 😛

Comment (david20571015) [reply torch/optim/lr_scheduler.py:203]: fixed

**Comment by david20571015** [comment]
Thanks, @janeyx99. I'm also wondering if this PR could be added to v2.3.1 by requiring a cherry-pick in the release tracker issue by myself. Is the release tracker issue available only for internal PyTorch team members or for every open-source contributor?

**Comment by janeyx99** [comment]
This wouldn’t qualify for the cherry-pick for 2.3.1 as this change isn’t a crucial fix for anything. Is there a reason to not wait for the next cycle 2.4?

**Comment by david20571015** [comment]
> This wouldn’t qualify for the cherry-pick for 2.3.1 as this change isn’t a crucial fix for anything. Is there a reason to not wait for the next cycle 2.4?

Sure, I agree with you. I just want to know if everyone can request a cherry-pick in the release tracker issue.

**Comment by janeyx99** [comment]
oh yes i believe so! https://dev-discuss.pytorch.org/t/pytorch-release-2-3-1-planning/2052

**Comment by david20571015** [comment]
> oh yes i believe so! https://dev-discuss.pytorch.org/t/pytorch-release-2-3-1-planning/2052

thanks for your reply 😊

**Pull request closed without merging**